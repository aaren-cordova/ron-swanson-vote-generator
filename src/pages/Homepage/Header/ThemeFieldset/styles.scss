@import 'styles/module';

.ThemeFieldset {
  border: none;
  display: flex;
  justify-content: center;
  align-items: center;
}

.ThemeFieldset__label {
  margin-right: rem(8px);

  &:first-child::after {
    content: ":";
  }
}