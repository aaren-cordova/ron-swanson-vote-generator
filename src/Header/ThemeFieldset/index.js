import classNames from 'classnames';
import './styles.scss';

export default function ThemeFieldset({className, theme, setTheme}) {
    function onChange(event) {
        setTheme(event.target.value);
    }

    return (
        <fieldset
            className={classNames('ThemeFieldset', className)}
            onChange={onChange}
        >
            <label className="ThemeFieldset__label">Theme</label>
            <input
                type="radio"
                id="theme-light-input"
                name="theme"
                value="light"
                checked={theme === 'light'}
            />
            <label
                className="ThemeFieldset__label"
                htmlFor="theme-light-input"
            >
                Light
            </label>
            <input
                type="radio"
                id="theme-dark-input"
                name="theme"
                value="dark"
                checked={theme === 'dark'}
            />
            <label
                className="ThemeFieldset__label"
                htmlFor="theme-dark-input"
            >
                Dark
            </label>
        </fieldset>
    );
}