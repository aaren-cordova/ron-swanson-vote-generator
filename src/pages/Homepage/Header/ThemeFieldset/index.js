import classNames from 'classnames';
import './styles.scss';
import PropTypes from 'prop-types';
import {Radio} from 'antd';

ThemeFieldset.propTypes = {
    className: PropTypes.string,
    setTheme: PropTypes.func.isRequired,
    theme: PropTypes.string.isRequired,
};
export default function ThemeFieldset({className, theme, setTheme}) {
    function onChange(event) {
        setTheme(event.target.value);
    }

    return (
        <fieldset
            className={classNames('ThemeFieldset', className)}
        >
            <label className="ThemeFieldset__label">Theme</label>

            <Radio.Group
                onChange={onChange}
                value={theme}
            >
                <Radio
                    id="theme-light-input"
                    name="theme"
                    value="light"
                />
                <label
                    className="ThemeFieldset__label"
                    htmlFor="theme-light-input"
                >
                    Light
                </label>
                <Radio
                    id="theme-dark-input"
                    name="theme"
                    value="dark"
                />
                <label
                    className="ThemeFieldset__label"
                    htmlFor="theme-dark-input"
                >
                    Dark
                </label>
            </Radio.Group>


        </fieldset>
    );
}