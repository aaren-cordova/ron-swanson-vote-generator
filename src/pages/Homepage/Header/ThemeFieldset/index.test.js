import renderer from 'react-test-renderer';
import ThemeFieldset from './index';

let props;

function render() {
    return <ThemeFieldset {...props}/>;
}

beforeEach(() => {
    props = {
        theme: '',
        setTheme: jest.fn(),
    };
});

test('matches the snapshot', () => {
    let component = renderer.create(render());
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('@theme="light" matches the snapshot', () => {
    props.theme = 'light';
    let component = renderer.create(render());
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('@theme="dark" matches the snapshot', () => {
    props.theme = 'dark';
    let component = renderer.create(render());
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
