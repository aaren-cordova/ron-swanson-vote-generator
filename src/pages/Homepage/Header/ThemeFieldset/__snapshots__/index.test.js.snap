// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@theme="dark" matches the snapshot 1`] = `
<fieldset
  className="ThemeFieldset"
>
  <label
    className="ThemeFieldset__label"
  >
    Theme
  </label>
  <div
    className="ant-radio-group ant-radio-group-outline"
  >
    <label
      className="ant-radio-wrapper"
    >
      <span
        className="ant-radio"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-radio-input"
          id="theme-light-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          type="radio"
          value="light"
        />
        <span
          className="ant-radio-inner"
        />
      </span>
    </label>
    <label
      className="ThemeFieldset__label"
      htmlFor="theme-light-input"
    >
      Light
    </label>
    <label
      className="ant-radio-wrapper ant-radio-wrapper-checked"
    >
      <span
        className="ant-radio ant-radio-checked"
        style={Object {}}
      >
        <input
          checked={true}
          className="ant-radio-input"
          id="theme-dark-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          type="radio"
          value="dark"
        />
        <span
          className="ant-radio-inner"
        />
      </span>
    </label>
    <label
      className="ThemeFieldset__label"
      htmlFor="theme-dark-input"
    >
      Dark
    </label>
  </div>
</fieldset>
`;

exports[`@theme="light" matches the snapshot 1`] = `
<fieldset
  className="ThemeFieldset"
>
  <label
    className="ThemeFieldset__label"
  >
    Theme
  </label>
  <div
    className="ant-radio-group ant-radio-group-outline"
  >
    <label
      className="ant-radio-wrapper ant-radio-wrapper-checked"
    >
      <span
        className="ant-radio ant-radio-checked"
        style={Object {}}
      >
        <input
          checked={true}
          className="ant-radio-input"
          id="theme-light-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          type="radio"
          value="light"
        />
        <span
          className="ant-radio-inner"
        />
      </span>
    </label>
    <label
      className="ThemeFieldset__label"
      htmlFor="theme-light-input"
    >
      Light
    </label>
    <label
      className="ant-radio-wrapper"
    >
      <span
        className="ant-radio"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-radio-input"
          id="theme-dark-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          type="radio"
          value="dark"
        />
        <span
          className="ant-radio-inner"
        />
      </span>
    </label>
    <label
      className="ThemeFieldset__label"
      htmlFor="theme-dark-input"
    >
      Dark
    </label>
  </div>
</fieldset>
`;

exports[`matches the snapshot 1`] = `
<fieldset
  className="ThemeFieldset"
>
  <label
    className="ThemeFieldset__label"
  >
    Theme
  </label>
  <div
    className="ant-radio-group ant-radio-group-outline"
  >
    <label
      className="ant-radio-wrapper"
    >
      <span
        className="ant-radio"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-radio-input"
          id="theme-light-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          type="radio"
          value="light"
        />
        <span
          className="ant-radio-inner"
        />
      </span>
    </label>
    <label
      className="ThemeFieldset__label"
      htmlFor="theme-light-input"
    >
      Light
    </label>
    <label
      className="ant-radio-wrapper"
    >
      <span
        className="ant-radio"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-radio-input"
          id="theme-dark-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          type="radio"
          value="dark"
        />
        <span
          className="ant-radio-inner"
        />
      </span>
    </label>
    <label
      className="ThemeFieldset__label"
      htmlFor="theme-dark-input"
    >
      Dark
    </label>
  </div>
</fieldset>
`;
